tests/storage/glusterfs/conftest.py

```diff
@@ -81,6 +82,45 @@ def gluster_disk(host, sr_disk_for_all_hosts):
         h.ssh(['umount', '/mnt/sr_disk'])
         h.ssh(['rmdir', '/mnt/sr_disk'])
 
+def _fallback_gluster_teardown(host):
+    # See: https://microdevsys.com/wp/volume-delete-volume-failed-some-of-the-peers-are-down/
+    # Remove all peers and bricks from the hosts volume and then stop and destroy volume.
+    def teardown_for_host(h):
+        print("< Fallback teardown on host: %s" % h)
+        hosts = h.pool.hosts
+
+        h.ssh(['systemctl', 'restart', 'glusterd'])
+
+        gluster_cmd = ['gluster', '--mode=script', 'volume', 'remove-brick', 'vol0', 'replica', '1']
+        for h2 in hosts:
+            if h.hostname_or_ip != h2.hostname_or_ip:
+                gluster_cmd.append('%s:/mnt/sr_disk/vol0/brick0' % h2.hostname_or_ip)
+        gluster_cmd.append('force')
+        h.ssh(gluster_cmd)
+
+        for h2 in hosts:
+            if h.hostname_or_ip != h2.hostname_or_ip:
+                h.ssh(['gluster', '--mode=script', 'peer', 'detach', h2.hostname_or_ip])
+
+        try:
+            # Volume might already be stopped if failure happened on delete
+            h.ssh(['gluster', '--mode=script', 'volume', 'stop', 'vol0'])
+        except Exception as e:
+            pass
+
+        h.ssh(['gluster', '--mode=script', 'volume', 'delete', 'vol0'])
+
+    try:
+        teardown_for_host(host)
+    except Exception as e:
+        print("< Fallback teardown failed on master: %s, attempting to teardown other hosts" % e)
+        for h in host.pool.hosts[1:]:
+            try:
+                teardown_for_host(h)
+            except Exception as e:
+                print("< Fallback teardown failed on host: %s with error: %s" % (h, e))
+                pass
+
```

Comment:
If even the fallback fails on master, I think we can just acknowledge that something went very wrong and stop here. The expected result of running the teardown on the host hosts after having maybe already half cleaned things up appears rather random to me.